File: jscomp/test/gpr_3209_test.js
'use strict';


function f9(param) {
  if (typeof param === "string") {
    switch (param) {
      case "T60" :
      case "T61" :
      case "T62" :
          return 1;
      default:
        return 3;
    }
  } else {
    switch (param.TAG) {
      case "T64" :
      case "T65" :
          return 2;
      default:
        return 3;
    }
  }
}

exports.f9 = f9;
/* No side effect */

File: jscomp/test/gpr_3209_test.ml
(* [@@@ocaml.warning "a-11-8-4-a"] *)
[@@@ocaml.warning "a+11+8-8"]

type t6 =
  | T60
  | T61
  | T62
  | T63
  | T64 of int
  | T65 of int
  | T66 of int
  | T68 of int

let f9 = function T60 | T61 | T62 -> 1 | T64 _ | T65 _ -> 2 | _ -> 3
